BlazarService/src/main/java/com/hubspot/blazar/listener/SlackRoomNotificationVisitor.java

```diff
@@ -147,4 +134,38 @@ private boolean shouldSend(InstantMessageConfiguration instantMessageConfigurati
     }
     return shouldSend;
   }
+
+  private void sendSlackMessageWithRetries(InstantMessageConfiguration configuration, SlackAttachment attachment, int tries) {
+    tries = tries - 1;
+    while (tries > 0) {
+      if (sendSlackMessage(configuration, attachment)) {
+        return;
+      } else {
+        tries--;
+      }
+    }
+    metricRegistry.counter("failed-slack-channel-sends").inc();
+  }
+
+  private boolean sendSlackMessage(InstantMessageConfiguration configuration, SlackAttachment attachment) {
```

Comment:
this looks about the same as the implementation in SlackImNotificationVisitor, maybe find a way to not have to duplicate the slack message sending code?